### PR TITLE
fix(source_slack): use private channels config when reading slack threads

### LINKED ISSUE #39414: [source-slack] Threads from private channels are not synced
### Connector Name

source-slack

### Connector Version

1.1.5

### What step the error happened?

During the sync

### Relevant information

I noticed that threads from private channels are not being ingested. Digging around a bit, I see that the `include_private_channels` configuration is not used when querying channels in the threads stream. The request for channels only specifies `public_channels`. 

It looks like a pretty simple fix, I'll submit a PR soon.

### Relevant log output

_No response_

### Contribute

- [X] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by ctdio**
## What

This fixes #39414. This PR allows for threads in private channels to be synced by using the existing `include_private_channels` config option. This option is used for consuming messages and channels from the declarative streams, but was not applied to the threads stream.

## How

This change passes in the `include_private_channels` configuration into the `Thread` constructor, which passes the value along to the `Channel`.  If `include_private_channels` is true, the `types` parameter in the `channels` request will be set to `private_channel,public_channel`, allowing private channels to be used when collecting threads.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by austinkelleher — APPROVED** [review]
LGTM! Looking forward to seeing this fix in place.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
@ctdio can you bump the connector version in the `metadata.yml`, `project.toml` and the doc changelog entry? After run `poetry lock` inside the connector folder to update the dependencies. After I'll trigger CI tests to validate your changes.

**Comment by ctdio** [comment]
Updated. Thanks @marcosmarxm!

File: poetry.lock [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.7.1 and should not be changed by hand.
+# This file is automatically @generated by Poetry 1.8.3 and should not be changed by hand.
```
Comment (marcosmarxm) [inline poetry.lock:1]: Not this file. You need to navigate inside the connector folder and run inside it.
```
cd airbyte-integrations/connectors/source-slack
poetry lock
```
Please revert this change and follow the steps above.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: poetry.lock (hunk 1 above)
Comment (ctdio) [reply poetry.lock:1]: Oh shoot, my bad. Fixing that now

Comment (ctdio) [reply poetry.lock]: Fixed @marcosmarxm. Thanks for catching that

**Comment by marcosmarxm** [comment]
Running CI tests let's wait for the results @ctdio

**Review by marcosmarxm — APPROVED** [review]
LGTM @ctdio only need an additional ✅ from the connector team as Slack is a certified connector.

**Review by natikgadzhi — APPROVED** [review]
<3 @ctdio thank you for the contribution! 

There are things to share ;)

We're migrating our sources to be as low-code as possible. In Slack, we only have this last stream (threads) that is in Python, the rest are manifest as you have seen. 

Here's the PR to migrate Slack Threads stream to yaml by @tolik0: https://github.com/airbytehq/airbyte/pull/38618. This PR in turn depends on the CDK PR https://github.com/airbytehq/airbyte/pull/38211 that's approved, and will likely be merged this week.

This Slack Threads → lowcode PR is a "breaking change", meaning users would have to re-sync their Threads stream when they upgrade to source Slack 2.0.0

With that in mind, we can do a few things here. 
1. We can merge this PR now, and then @tolik0 will add private channel threads support to the low-code one before releasing #38618. 
2. We can skip this PR and just ask @tolik0 to add privae channels support to his PR.

Since this looks clean, I'm not opposed to merging this now, so — LGTM from where I sit.

**Comment by natikgadzhi** [comment]
@irvingpop, looky here! Getting closer.

**Comment by ctdio** [comment]
Hey @natikgadzhi, thanks for taking a look at this! I would appreciate it if we could go with option 1 and release this before migrating to the low-code implementation.

We are hoping this could land in production sooner rather than later since it's a pretty severe gap in functionality we've encountered that's affecting business operations for us (we're using Airbyte for a number of customer integrations).

**Comment by marcosmarxm** [comment]
Hello @ctdio  your feedback matters a lot to us. Can you spare just 3 minutes to complete a [survey](https://form.typeform.com/to/DOBqejte)? We’re dedicated to making the contributor experience even better, and your input is key to achieving excellence. Thank you for helping us improve!

**Comment by ctdio** [comment]
Yes, happy to! Thank you, @marcosmarxm and @natikgadzhi, for helping me get this through!
